pkg/apiserver/authentication/token/jwt.go

```diff
@@ -54,35 +60,49 @@ func (s *jwtTokenIssuer) Verify(tokenString string) (User, error) {
 	}
 
 	clm := &Claims{}
+
+	// verify token signature and expiration time
 	_, err := jwt.ParseWithClaims(tokenString, clm, s.keyFunc)
+
 	if err != nil {
 		return nil, err
 	}
 
-	// accessTokenMaxAge = 0 or token without expiration time means that the token will not expire
-	// do not validate token cache
-	if s.options.OAuthOptions.AccessTokenMaxAge > 0 && clm.ExpiresAt > 0 {
-		_, err = s.cache.Get(tokenCacheKey(tokenString))
-
+	// If multiple login is not enabled, need to verify whether the token has been revoked
+	if !s.options.MultipleLogin {
+		err = s.validate(clm.Username, tokenString)
 		if err != nil {
-			if err == cache.ErrNoSuchKey {
-				return nil, errTokenExpired
-			}
+			klog.Error(err)
 			return nil, err
 		}
 	}
 
 	return &user.DefaultInfo{Name: clm.Username, UID: clm.UID}, nil
 }
 
+func (s *jwtTokenIssuer) validate(username string, token string) error {
+	loginRecords, err := s.loginRecordInformer.Lister().List(labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: username}))
```

Comment:
I don't it's a good idea to check login records here. The only responsibility of TokenIssuer is to deal with token alone.